lib/plugins/wallet/monerod/monerod.js

```diff
@@ -0,0 +1,206 @@
+const E = require('../../../error')
+const { logger } = require('../../../blockchain/common')
+
+const cryptoRec = utils.getCryptoCurrency('XMR')
+const configPath = utils.configPath(cryptoRec, blockchainUtils.blockchainDir())
+const walletDir = path.resolve(utils.cryptoDir(cryptoRec, blockchainUtils.blockchainDir()), 'wallets')
+const unitScale = cryptoRec.unitScale
+
+function rpcConfig () {
+  try {
+    const config = jsonRpc.parseConf(configPath)
+    return {
+      username: config['rpc-login'].split(':')[0],
+      password: config['rpc-login'].split(':')[1],
+      port: cryptoRec.walletPort || cryptoRec.defaultPort
+    }
+  } catch (err) {
+    throw new Error('wallet is currently not installed')
+  }
+}
+
+function fetch (method, params) {
+  return jsonRpc.fetchDigest(rpcConfig(), method, params)
+}
+
+function handleError (error) {
+  switch(error.code) {
+    case -13:
+      {
+        if (
+          fs.existsSync(path.resolve(walletDir, 'Wallet')) &&
+          fs.existsSync(path.resolve(walletDir, 'Wallet.keys')) &&
+          fs.existsSync(path.resolve(walletDir, 'Wallet.address.txt'))
+        ) {
+          logger.debug('Found wallet! Opening wallet...')
+          return openWallet()
+        }
+        logger.debug('Couldn\'t find wallet! Creating...')
+        return createWallet()
+      }
+    case -21:
+      throw new Error('Wallet already exists!')
+    case -17:
+      throw new E.InsufficientFundsError()
+    case -37:
+      throw new E.InsufficientFundsError()
+    default:
+      throw new Error(
+        _.join(' ', [
+          'json-rpc::got error:',
+          JSON.stringify(_.get('message', error, '')),
+          JSON.stringify(_.get('response.data.error', error, ''))
+        ])
+      )
+  }
+}
+
+function openWallet () {
+  return fetch('open_wallet', { filename: 'Wallet', password: rpcConfig().password })
+}
+
+function createWallet () {
+  return fetch('create_wallet', { filename: 'Wallet', password: rpcConfig().password, language: 'English' })
+    .then(() => new Promise(() => setTimeout(() => openWallet(), 3000)))
+    .then(() => fetch('auto_refresh'))
+}
+
+function checkCryptoCode (cryptoCode) {
+  if (cryptoCode !== 'XMR') return Promise.reject(new Error('Unsupported crypto: ' + cryptoCode))
+  return Promise.resolve()
+}
+
+function refreshWallet () {
+  return fetch('refresh')
+}
+
+function accountBalance (cryptoCode) {
+  return checkCryptoCode(cryptoCode)
+    .then(() => refreshWallet())
+    .then(() => fetch('get_balance', { account_index: 0, address_indices: [0] }))
+    .then(res => {
+      return BN(res.unlocked_balance).shift(unitScale).round()
+    })
+    .catch(err => handleError(err))
+}
+
+function balance (account, cryptoCode) {
+  return accountBalance(cryptoCode)
+}
+
+function sendCoins (account, address, cryptoAtoms, cryptoCode) {
+  return checkCryptoCode(cryptoCode)
+    .then(() => refreshWallet())
+    .then(() => fetch('transfer_split', {
```

Comment:
is there a reason to use `transfer_split` instead of `transfer`

Reply:
From what I read, both endpoints do the same, but `transfer_split` splits a transaction into multiple smaller ones, if needed. I believe `transfer_split` is similar to a `transfer` 2.0, and act the same on regular, single-destination transactions and has improvements on multi-destination transactions, which we could implement in the future.
So yeah, for those reasons, I chose `transfer_split`